src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -0,0 +1,86 @@
+package mesosphere.marathon.api
+
+import java.net.URI
+import javax.servlet.http.{ HttpServletResponse, HttpServletRequest }
+import javax.ws.rs.core.Response.Status
+
+import mesosphere.marathon.core.appinfo.AppSelector
+import mesosphere.marathon.interface.auth._
+import mesosphere.marathon.interface.http.{ HttpRequest, HttpResponse }
+import mesosphere.marathon.state.{ PathId, AppDefinition }
+import scala.collection.JavaConverters._
+import javax.ws.rs.core.{ NewCookie, Response }
+
+/**
+  * Base trait for authentication and authorization in http resource endpoints.
+  */
+trait AuthResource extends RestResource {
+
+  def authenticator: Authenticator
+  def authorizer: Authorizer
+
+  def doIfAuthenticated(request: HttpServletRequest,
+                        response: HttpServletResponse)(fn: Identity => Response): Response = {
+    val requestWrapper = new RequestFacade(request)
+    val identity = result(authenticator.authenticate(requestWrapper))
+    identity.map(fn).getOrElse {
+      val responseWrapper = new ResponseFacade
+      authenticator.handleNotAuthenticated(requestWrapper, responseWrapper)
+      responseWrapper.response
+    }
+  }
+
+  def doIfAuthorized[Resource](request: HttpServletRequest,
+                               response: HttpServletResponse,
+                               action: AuthorizedAction[Resource],
+                               resources: Resource*)(fn: Identity => Response): Response = {
+    def isAllowed(id: Identity) = resources.forall(authorizer.isAuthorized(id, action, _))
+    def response(fn: HttpResponse => Unit): Response = {
+      val responseFacade = new ResponseFacade
+      fn(responseFacade)
+      responseFacade.response
+    }
+    val requestFacade = new RequestFacade(request)
+    result(authenticator.authenticate(requestFacade)) match {
```

Comment:
new line? I also like a new line between functions, but that's just my personal taste.